File: content/development/golang-first-impression/private.rb
#!/usr/bin/env ruby

class Sample
  def public_method
    puts "this is public"
  end

  def call_private
    private_method
  end

  private
  def private_method
    puts "this is private"
  end
end

s = Sample.new
s.public_method
s.call_private
begin
  s.private_method
rescue NoMethodError => e
  puts "Error: #{e}"
end

File: content/development/golang-first-impression/private/private.go
package private

import "fmt"

type Sample struct{}

func (s *Sample) PublicMethod() {
	fmt.Println("this is public")
}

func (s *Sample) privateMethod() {
	fmt.Println("this is private")
}

func (s *Sample) CallPrivate() {
	s.privateMethod()
}
